File: code/community/1304597127434866691/improve-speech-recognition-noisy-environments.py
import os
from deepgram import Deepgram
import asyncio

async def main():
    # Load environment variables
    api_key = os.getenv("DEEPGRAM_API_KEY")
    audio_url = os.getenv("AUDIO_URL")

    # Create Deepgram SDK instance
    dg_client = Deepgram(api_key)

    # Set the options for transcription with noise filtering
    options = {"punctuate": True, "noise_reduction": True}

    try:
        # Transcribe the audio from the URL
        response = await dg_client.transcription.prerecorded({"url": audio_url}, options)
        
        # Print the transcription
        print(response['results']['channels'][0]['alternatives'][0]['transcript'])
    except Exception as e:
        print(f"An error occurred: {e}")

# Run the transcription task
asyncio.run(main())

File: code/community/1304597127434866691/improve-speech-recognition-noisy-environments.js
require('dotenv').config();
const { Deepgram } = require('@deepgram/sdk');

const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const audioUrl = process.env.AUDIO_URL;

const deepgram = new Deepgram(deepgramApiKey);

(async () => {
  try {
    const response = await deepgram.transcription.preRecorded({
      url: audioUrl,
      options: {
        punctuate: true,
        noiseReduction: true // Enable noise reduction
      },
    });
    
    console.log('Transcription:', response.results.channels[0].alternatives[0].transcript);
  } catch (err) {
    console.error('Error:', err);
  }
})();